app/gradescope_to_spreadsheet.py

```diff
@@ -0,0 +1,334 @@
+    # Save the credentials for the next run
+    with open("token.json", "w") as token:
+        token.write(creds.to_json())
+    return creds
+
+@deprecated
+def writeToSheet(sheet_api_instance, assignment_scores, assignment_id = ASSIGNMENT_ID):
+    try:
+        sub_sheet_titles_to_ids = get_sub_sheet_titles_to_ids(sheet_api_instance)
+
+        sheet_id = None
+
+        if assignment_id not in sub_sheet_titles_to_ids:
+            create_sheet_request = {
+                "requests": {
+                    "addSheet": {
+                        "properties": {
+                            "title": assignment_id
+                        }
+                    }
+                }
+            }
+            response = sheet_api_instance.batchUpdate(spreadsheetId=SPREADSHEET_ID, body=create_sheet_request).execute()
+            sheet_id = response['replies'][0]['addSheet']['properties']['sheetId']
+        else:
+            sheet_id = sub_sheet_titles_to_ids[assignment_id]
+        time.sleep(5)
+        update_sheet_with_csv(assignment_scores, sheet_api_instance, sheet_id)
+        print("Successfully updated spreadsheet with new score data")
+    except HttpError as err:
+        print(err)
+
+@deprecated
+def create_sheet_api_instance(creds):
+    service = build("sheets", "v4", credentials=creds)
+    sheet_api_instance = service.spreadsheets()
+    return sheet_api_instance
+
+@deprecated
+def get_sub_sheet_titles_to_ids(sheet_api_instance):
+    global subsheet_titles_to_ids
+    if subsheet_titles_to_ids:
+        return subsheet_titles_to_ids
+    sheets = sheet_api_instance.get(spreadsheetId=SPREADSHEET_ID, fields='sheets/properties').execute()
+    sub_sheet_titles_to_ids = {sheet['properties']['title']: sheet['properties']['sheetId'] for sheet in
+                               sheets['sheets']}
+    return sub_sheet_titles_to_ids
+
+@deprecated
+def update_sheet_with_csv(assignment_scores, sheet_api_instance, sheet_id, rowIndex = 0, columnIndex=0):
+    push_grade_data_request = {
+        'requests': [
+            {
+                'pasteData': {
+                    "coordinate": {
+                        "sheetId": sheet_id,
+                        "rowIndex": rowIndex,
+                        "columnIndex": columnIndex,
+                    },
+                    "data": assignment_scores,
+                    "type": 'PASTE_NORMAL',
+                    "delimiter": ',',
+                }
+            }
+        ]
+    }
+    sheet_api_instance.batchUpdate(spreadsheetId=SPREADSHEET_ID, body=push_grade_data_request).execute()
+
+@deprecated
+def retrieve_grades_from_gradescope(gradescope_client, assignment_id = ASSIGNMENT_ID):
+    assignment_scores = str(gradescope_client.download_scores(COURSE_ID, assignment_id)).replace("\\n", "\n")
+    return assignment_scores
+
+@deprecated
+def initialize_gs_client():
+    gradescope_client = client.GradescopeClient()
+    gradescope_client.prompt_login()
```

Comment:
Please pull the latest changes from main. Naveen changed this line of code.

Reply:
And then push to your branch again.

Reply:
Make sure you have a `.env` to load the GRADESCOPE_EMAIL and GRADESOPE_PASSWORD.